### PR TITLE
run more examples as part of GitHub Actions

### PR DESCRIPTION
**Pull request by wesk**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: examples/launch_tuning_gluonts.py
Comment (geoalgo) [inline examples/launch_tuning_gluonts.py:91]: Why setting a new constant here? Setting 3000 seconds is fine with me but the comment is then not in sync.

File: examples/launch_huggingface_classification.py
Comment (geoalgo) [inline examples/launch_huggingface_classification.py:80]: Why setting a new constant here? Setting 3000 seconds is fine with me but the comment is then not in sync.

File: .github/workflows/run-syne-tune.yml
Comment (geoalgo) [inline .github/workflows/run-syne-tune.yml:56]: The naming `Run Syne Tune` could perhaps be made more precise and self-revealing, perhaps `Run Syne Tune script` is already a bit more telling given that you use script path?

File: .github/workflows/fast-tests.yml
Comment (geoalgo) [inline .github/workflows/fast-tests.yml:1]: Could you define "fast"? For instance, "Evaluate fast tests that runs under 5 minutes" as fast can be quite subjective.

File: .github/workflows/integ-tests.yml
Comment (geoalgo) [inline .github/workflows/integ-tests.yml:11]: Fine for now, but I suppose we will want to limit the frequency of those (for instance weekly in addition to release) as they will be a tad expensive.

File: .github/workflows/integ-tests.yml
Comment (geoalgo) [inline .github/workflows/integ-tests.yml:31]: Great work and great that you think of this case.

File: .github/workflows/integ-tests.yml
Comment (geoalgo) [inline .github/workflows/integ-tests.yml:33]: This should probably be more inclusive, for instance it should probably also include "Starting".

File: .github/workflows/integ-tests.yml
Comment (geoalgo) [inline .github/workflows/integ-tests.yml:90]: This is fine with me as is, but why not always adding `needs: stop_all_training_jobs`?
It does cost if no job needs to be stopped but it makes it allow to not forget to add this line (which can happen if you think you dont need it whereas the examples actually spawed SM jobs).

File: .github/workflows/integ-tests.yml
Comment (geoalgo) [inline .github/workflows/integ-tests.yml:119]: Isnt this one fast?

Ok I see that some of the test appear in both this action and the "fast" action, what is the rationale for running the test twice?

**Review by geoalgo — COMMENTED** [review]
Very nice work!

File: examples/launch_tuning_gluonts.py
Comment (wesk) [reply examples/launch_tuning_gluonts.py:91]: This is to ensure that the example finishes in less than one hour. The problem was that the example was failing due to credential expiration (the creds are only valid for one hour). If the experiment runs for an hour, then the runtime goes slightly over an hour because there are a few seconds of setup time before the test reaches the example script. 

The comment is for a future reader who might want to run it for longer

File: examples/launch_huggingface_classification.py
Comment (wesk) [reply examples/launch_huggingface_classification.py:80]: Same rationale as previous instance

File: .github/workflows/fast-tests.yml
Comment (wesk) [reply .github/workflows/fast-tests.yml:1]: What I really mean is that all tests in this file must satisfy the following two properties: 
1. We want them to run against every PR
2. They do not require AWS credentials (We only run the tests that require credentials on code that has been approved and merged to main)

"Fast" seemed a reasonable way of referring to them, but some possible other names are: 
1. Integ Tests (and we then rename the "Integ Tests" suite to "End-to-end Tests", since that's more of what that other suite is).
2. Lightweight Integ Tests (in contrast with the other "Full integ tests")

If we go with this file becoming "Integ Tests" and the other one becoming "End-to-end Tests", then we can trigger this one with every PR and merge, and trigger the other one with every merge to `main`, and remove the duplicated tests between the two.

File: .github/workflows/integ-tests.yml
Comment (wesk) [reply .github/workflows/integ-tests.yml:11]: Before merging this PR, I plan to push one more commit removing these lines. Then, this suite will run 1) on merges to `main` and 2) on-demand if someone with write access to the repository clicks the button to run it from the GitHub UI.

This way, we will know as soon as a bad commit is merged to main. Whereas if we only run it before a release, we might not catch a regression in main for a long time. I think at least until we have a consistent release schedule, it makes sense to start by running them against main and seeing how well it works for us.

File: .github/workflows/integ-tests.yml
Comment (wesk) [reply .github/workflows/integ-tests.yml:90]: Good point, there's some duplication of tests between this file and the other one. I think we could do the following: 
1. Rename this file to be "End-to-end tests"
2. Remove all tests that are in the "Fast tests" from this file
3. All remaining tests in this file will always need the `needs: stop_all_training_jobs` line in this file

This way the role of each test suite is more clear

See this earlier comment also for more of what I'm thinking now, based on your comment: https://github.com/awslabs/syne-tune/pull/520#discussion_r1095523249

File: .github/workflows/integ-tests.yml
Comment (wesk) [reply .github/workflows/integ-tests.yml:119]: Good point, see comment https://github.com/awslabs/syne-tune/pull/520#discussion_r1095533413 for proposed fix

File: .github/workflows/integ-tests.yml
Comment (wesk) [reply .github/workflows/integ-tests.yml:33]: I spent a few minutes and I haven't found the syntax for how to match two statues from the AWS CLI, do you know what it would look like? In my testing, I haven't found it to be a problem, since we have enough of a limit in the account to have some buffer

File: .github/workflows/fast-tests.yml
Comment (geoalgo) [reply .github/workflows/fast-tests.yml]: Either name is fine, can you add a comment explaining the rationale you just described so that the context is clear for others? (for instance saying that this is a list of tests that dont require cred and runs at all PRs)

File: .github/workflows/integ-tests.yml
Comment (geoalgo) [reply .github/workflows/integ-tests.yml]: Just checked and it cannot be done indeed.

**Review by geoalgo — APPROVED** [review]

File: .github/workflows/fast-tests.yml
Comment (wesk) [reply .github/workflows/fast-tests.yml]: Done, updated in the next revision

**Review by geoalgo — APPROVED** [review]

**Pull request merged**
